x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/Clamp.java

```
@@ -0,0 +1,153 @@
+        TypeResolution resolution = TypeResolutions.isType(
+            field,
+            t -> t.isNumeric() || t == DataType.BOOLEAN || t.isDate() || DataType.isString(t) || t == DataType.IP || t == DataType.VERSION,
+            sourceText(),
+            TypeResolutions.ParamOrdinal.FIRST,
+            fieldDataType.typeName()
+        );
+        if (resolution.unresolved()) {
+            return resolution;
+        }
+        if (fieldDataType == NULL) {
+            return new TypeResolution("'field' must not be null in clamp()");
+        }
+        for (Expression child : List.of(children().get(1), children().get(2))) {
+            var childRes = TypeResolutions.isType(
+                child,
+                t -> t.isNumeric() ? fieldDataType.isNumeric() : t.noText() == fieldDataType,
+                sourceText(),
+                child == children().get(1) ? TypeResolutions.ParamOrdinal.SECOND : TypeResolutions.ParamOrdinal.THIRD,
+                fieldDataType.typeName()
+            );
+            if (childRes.unresolved()) {
+                return childRes;
+            }
+        }
+        return TypeResolution.TYPE_RESOLVED;
+    }
+
+    @Override
+    public DataType dataType() {
+        return field.dataType();
+    }
+
+    @Override
+    public Expression replaceChildren(List<Expression> newChildren) {
+        return new Clamp(source(), newChildren.get(0), newChildren.get(1), newChildren.get(2));
+    }
+
+    @Override
+    protected NodeInfo<? extends Expression> info() {
+        return NodeInfo.create(this, Clamp::new, field, children().get(1), children().get(2));
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        source().writeTo(out);
+        out.writeNamedWriteable(field);
+        out.writeNamedWriteable(min);
+        out.writeNamedWriteable(max);
+    }
+
+    @Override
+    public ExpressionEvaluator.Factory toEvaluator(ToEvaluator toEvaluator) {
+        return null;
```

can we throw UnsupportedOperationException instead?